### DISCUSSION TITLE
Best practices to deal with log truncation

### DISCUSSION THREAD

**Discussion by alfonso-paella** [vercel/community#1287] [Help]
### Summary

We use NextJS on Vercel, with Datadog for logging.

We're frustrated by a limitation of Vercel, which is that logs forwarded to Datadog are capped at 4KB per log. Because of this, there's countless times where we have not been able to get any repro info for critical issues, as sometimes a stack trace took up all the available step.

Has anyone found a workaround or best practice to avoid this issue?

**Comment by buildgreatthings** [comment]
I second this!

**Comment by smaeda-ks** [comment]
Another option would be to emit logs from your functions:
https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30#agentless-logging
